### PR TITLE
GetTotalPauseDuration implementation (#68835)

### PR DESCRIPTION
**Pull request by cshung**
Backport of https://github.com/dotnet/runtime/pull/68835 to release/6.0-staging

## Customer Impact
One of our first-party customers reported that they need `GC.GetTotalPauseDuration` to help with their autoscaling. Here is what he said:

We need `GC.GetTotalPauseDuration` in order to *accurately* measure the impact of GC in our services and in order for us to have actionable signals to manage autoscaling policies. We have faced numerous customer-impacting Sev2 incidents where a .NET process was CPU-limited yet not consuming 100% CPU, which in several distinct cases we concluded was due to Workstation GC pauses. With `GC.GetTotalPauseDuration`, we will be able to accurately measure how close we are to hitting CPU limits, which enables us to configure appropriate autoscaling policies that would help us avoid customer impact. While the existing EventCounter “time-in-gc” helps to some extent, it is a noisy and unreliable source because it doesn’t measure the actual average time in GC, just the average time in GC for the most recent GC, which is skewed since not all GC’s are comparable (different gen’s, compacting Vs. not, etc.)

## Testing
- [x] GC Perf Sim 
	- [x] Normal Server 
	- [x] Normal Workstation 
	- [x] Large Pages Server 
	- [x] Large Pages Workstation 
	- [x] Low Memory Container 
	- [x] High Memory Load
- [x] Microbenchmarks 
	- [x] V8 Test 
	- [x] Top 20 Microbenchmarks
- [x] ASPNet Benchmarks 
	- [x] JsonMin - Windows 

## Risk
Low, this is an additional API. Any existing code that does not call the new API should not be affected at all.

### CODE REVIEW

**Comment by carlossanlop** [comment]
@cshung I see this is still a draft. Today is code complete for the July release. If you want this change to be included in that release, please get a tactics approval, a sign-off, confirm the CI failures are unrelated, and merge it before EOD today.

**Comment by carlossanlop** [comment]
@cshung I see this is still a draft. Reminder - Tomorrow Monday 10th is Code Complete for the August Release. If you intend to get this fix included in that servicing release, please make sure to get a Tactics approval and merge the PR before 4pm, because that's when I close the branches to start merging staging into internal.

**Review by jeffschwMSFT — APPROVED** [review]
approved.  please get a code review.  we will take for consideration in 6.0.x

File: src/libraries/System.Runtime/ref/System.Runtime.cs [hunk 1]
```diff
@@ -2950,6 +2950,12 @@ public static partial class GC
         public static System.GCNotificationStatus WaitForFullGCComplete() { throw null; }
         public static System.GCNotificationStatus WaitForFullGCComplete(int millisecondsTimeout) { throw null; }
         public static void WaitForPendingFinalizers() { }
+
+        /// <summary>
+        /// Gets the total amount of time paused in GC since the beginning of the process.
+        /// </summary>
+        /// <returns> The total amount of time paused in GC since the beginning of the process.</returns>
+        public static TimeSpan GetTotalPauseDuration() { return TimeSpan.Zero; }
```
[the remark this answers is not in the document]
Comment (cshung) [reply src/libraries/System.Runtime/ref/System.Runtime.cs]: Yes, the warning shows up. How do I suppress them?

[the remark this answers is not in the document]
Comment (cshung) [reply src/libraries/System.Runtime/ref/System.Runtime.cs]: @ericstj, can you please help?

[the remark this answers is not in the document]
Comment (ericstj) [reply src/libraries/System.Runtime/ref/System.Runtime.cs]: Sorry have been in meetings, having a look now.

File: src/libraries/System.Runtime/src/MatchingRefApiCompatBaseline.txt [hunk 2]
```diff
@@ -24,4 +24,5 @@ CannotMakeMemberAbstract : Member 'public System.Boolean System.IO.FileSystemInf
 CannotMakeMemberAbstract : Member 'public System.String System.IO.FileSystemInfo.Name.get()' is abstract in the reference but is not abstract in the implementation.
 # C# generates backing fields for fixed buffers as public.
 TypesMustExist : Type 'System.IO.Enumeration.FileSystemEntry.<_fileNameBuffer>e__FixedBuffer' does not exist in the reference but it does exist in the implementation.
-
+# This API only exists on servicing builds after the commit that introduce it
+MembersMustExist : Member 'public System.TimeSpan System.GC.GetTotalPauseDuration() does not exist in the reference but it does exist in the implementation.
```
Comment (ericstj) [inline src/libraries/System.Runtime/src/MatchingRefApiCompatBaseline.txt:28]:
```suggestion
MembersMustExist : Member 'public System.TimeSpan System.GC.GetTotalPauseDuration()' does not exist in the reference but it does exist in the implementation.
```

File: src/tests/GC/API/GC/GetTotalPauseDuration.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+public class Test_Collect
+{
+    public static int Main()
+    {
+        Stopwatch sw = Stopwatch.StartNew();
+        GC.Collect();
+        sw.Stop();
+        TimeSpan elapsed = sw.Elapsed;
+        TimeSpan totalPauseDuration = GC.GetTotalPauseDuration();
```
Comment (ericstj) [inline src/tests/GC/API/GC/GetTotalPauseDuration.cs:16]: Looks like this is failing on mono.  You might need to ensure that this test is either building against System.Private.CoreLib or calling this API through reflection.

**Comment by jeffschwMSFT** [comment]
no-merge can be removed once review feedback addressed

**Review by Maoni0 — APPROVED** [review]
LGTM!

**Review by ericstj — APPROVED** [review]

**Review by mangod9 — APPROVED** [review]

**Pull request merged**